### ISSUE TITLE
Geny integration

### ISSUE THREAD

**Issue by szymon-rd** [softwaremill/sttp#1822]
Integration with [geny](https://github.com/com-lihaoyi/geny) could allow STTP to work with [os-lib](https://github.com/com-lihaoyi/os-lib). It would be beneficial to use these two together within the toolkit.

**Comment by adamw** [comment]
So that you could upload a file that's defined in terms of os-lib abstractions, for example?

**Comment by szymon-rd** [comment]
Yes, exactly. That os-lib's path could be provided as body. We could also support streaming to/from files via geny integration and `java.io.OutputStream`, as I think that's the way the write streaming is handled in os-lib.
